nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/main/java/org/nd4j/parameterserver/distributed/logic/v2/FileChunksTracker.java

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ * Copyright (c) 2015-2018 Skymind, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.parameterserver.distributed.logic.v2;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.val;
+import org.apache.commons.lang3.SerializationUtils;
+import org.nd4j.linalg.exception.ND4JIllegalStateException;
+import org.nd4j.linalg.primitives.AtomicBoolean;
+import org.nd4j.parameterserver.distributed.messages.v2.VoidChunk;
+import org.nd4j.parameterserver.distributed.messages.v2.VoidMessage_v2;
+
+import java.io.*;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * File-based implementation of ChunksTracker
+ */
+public class FileChunksTracker<T extends VoidMessage_v2> implements ChunksTracker<T> {
+    @Getter
+    private final String originId;
+
+    private final int numChunks;
+
+    private File holder;
+
+    private Map<Integer, AtomicBoolean> map = new ConcurrentHashMap<>();
+
+    public FileChunksTracker(VoidChunk chunk) {
+        originId = chunk.getOriginalId();
+        numChunks = chunk.getNumberOfChunks();
+        try {
+            holder = File.createTempFile("FileChunksTracker", "Message");
```

Comment:
Let's make dir configurable. User might want it on SSD different to system drive.

Reply:
Yes, once we'll be ready to merge lots of stuff will become configurable.